src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -874,6 +909,24 @@ public void AddRequestContext()
                 _parameters.Add(KnownParameters.RequestContext);
             }
 
+            private void AddRequestParameterWithMediaTypes(RequestParameter requestParameter, ICollection<string> requestMediaTypes)
+            {
+                CSharpType type = new CSharpType(typeof(ContentType), requestParameter.IsNullable || !requestParameter.IsRequired);
+                var parameter = new Parameter(
+                    requestParameter.CSharpName(),
+                    CreateDescriptionWithMediaTypes(requestParameter, requestMediaTypes),
+                    TypeFactory.GetInputType(type),
+                    null,
+                    requestParameter.IsRequired,
+                    IsApiVersionParameter: false,
+                    IsResourceIdentifier: requestParameter.IsResourceParameter,
+                    SkipUrlEncoding: requestParameter.Extensions?.SkipEncoding ?? false,
```

Comment:
We can use default values for `IsResourceIdentifier` and `SkipUrlEncoding`, so whole constructor call should look like this:
```cs
var name = requestParameter.CSharpName();
var description = CreateDescriptionWithMediaTypes(requestParameter, requestMediaTypes);
var parameter = new Parameter(name, description, typeof(ContentType), null, true, RequestLocation: RequestLocation.Body);
```

Reply:
Is there any possibility that `requestParameter.IsRequired` is false to `ContentType`?

Reply:
It doesn't really matter. We need to specify content type in our protocol methods. And since `ContentType` is a struct, we don't have to validate this parameter for null.

The question, however, is should we validate specific values here. Should we check if user specifies content type that is not supported by the service.

Reply:
> Should we check if user specifies content type that is not supported by the service.

No, it'd be better to let the service validate.  Ideally they'd send back a clear/obvious 415, but most of Azure is JSON so it I don't expect serious discoverability problems here.

Reply:
Then we don't need any additional validation here at all.